### DISCUSSION TITLE
Whisper (local) in STT Setting not working

### DISCUSSION THREAD

**Discussion by edisonzf2020** [open-webui/open-webui#1996] [General]
# Bug Report

## Description
choice Whisper(local) in STT Setting , but It's not working when use mic for voice input.

<img width="927" alt="截屏2024-05-06 11 17 14" src="https://github.com/open-webui/open-webui/assets/95912883/ee3a9e1c-7518-446c-a6f8-a34f91df847a">

Check the data/cache directory and no automatic download of whisper is found.

![image](https://github.com/open-webui/open-webui/assets/95912883/65087045-21d8-4731-a560-4d4dee69c6f4)

Docker deployment uses the 'WHISPER_MODEL_AUTO_UPDATE=True' parameter

![image](https://github.com/open-webui/open-webui/assets/95912883/88616adb-4ec0-474a-8567-bc21b3354bbf)

**Bug Summary:**
[Provide a brief but clear summary of the bug]

**Steps to Reproduce:**
[Outline the steps to reproduce the bug. Be as detailed as possible.]

**Expected Behavior:**
[Describe what you expected to happen.]

**Actual Behavior:**
[Describe what actually happened.]

## Environment

- **Open WebUI Version:** [e.g., 0.1.120]. v0.1.123

- **Ollama (if applicable):** [e.g., 0.1.30, 0.1.32-rc1]  0.1.33

- **Operating System:** [e.g., Windows 10, macOS Big Sur, Ubuntu 20.04]
- **Browser (if applicable):** [e.g., Chrome 100.0, Firefox 98.0]

## Reproduction Details

**Confirmation:**

- [ ] I have read and followed all the instructions provided in the README.md.
- [ ] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.

## Logs and Screenshots

**Browser Console Logs:**
[Include relevant browser console logs, if applicable]

**Docker Container Logs:**
[Include relevant Docker container logs, if applicable]

**Screenshots (if applicable):**
[Attach any relevant screenshots to help illustrate the issue]

## Installation Method

[Describe the method you used to install the project, e.g., manual installation, Docker, package manager, etc.]

## Additional Information

[Include any additional details that may help in understanding and reproducing the issue. This could include specific configurations, error messages, or anything else relevant to the bug.]

## Note

If the bug report is incomplete or does not follow the provided instructions, it may not be addressed. Please ensure that you have followed the steps outlined in the README.md and troubleshooting.md documents, and provide all necessary information for us to reproduce and address the issue. Thank you!

**Comment by edisonzf2020** [comment]
If this isn't a bug, is there something I'm missing in my setup?

**Comment by edisonzf2020** [comment]
I reinstalled the docker file, and the log shows that an error was indeed reported when downloading whisper. It must be a bug, right?
The log is as follows:
```
No WEBUI_SECRET_KEY provided
Generating WEBUI_SECRET_KEY
Loading WEBUI_SECRET_KEY from .webui_secret_key
/usr/local/lib/python3.11/site-packages/pydantic/_internal/_fields.py:160: UserWarning: Field "model_name" has conflict with protected namespace "model_".
You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
  warnings.warn(
/usr/local/lib/python3.11/site-packages/pydantic/_internal/_fields.py:160: UserWarning: Field "model_info" has conflict with protected namespace "model_".
You may be able to resolve this warning by setting `model_config['protected_namespaces'] = ()`.
  warnings.warn(
Cannot determine model snapshot path: Cannot find an appropriate cached snapshot folder for the specified revision on the local disk and outgoing traffic has been disabled. To enable repo look-ups and downloads online, pass 'local_files_only=False' as input.
Traceback (most recent call last):
  File "/app/backend/apps/rag/utils.py", line 391, in get_model_path
    model_repo_path = snapshot_download(**snapshot_kwargs)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huggingface_hub/utils/_validators.py", line 114, in _inner_fn
    return fn(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/huggingface_hub/_snapshot_download.py", line 220, in snapshot_download
    raise LocalEntryNotFoundError(
huggingface_hub.utils._errors.LocalEntryNotFoundError: Cannot find an appropriate cached snapshot folder for the specified revision on the local disk and outgoing traffic has been disabled. To enable repo look-ups and downloads online, pass 'local_files_only=False' as input.
/usr/local/lib/python3.11/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
/usr/local/lib/python3.11/site-packages/huggingface_hub/file_download.py:1132: FutureWarning: `resume_download` is deprecated and will be removed in version 1.0.0. Downloads always resume when possible. If you want to force a new download, use `force_download=True`.
  warnings.warn(
INFO:     Started server process [1]
INFO:     Waiting for application startup.
  ___                    __        __   _     _   _ ___ 
 / _ \ _ __   ___ _ __   \ \      / /__| |__ | | | |_ _|
| | | | '_ \ / _ \ '_ \   \ \ /\ / / _ \ '_ \| | | || | 
| |_| | |_) |  __/ | | |   \ V  V /  __/ |_) | |_| || | 
 \___/| .__/ \___|_| |_|    \_/\_/ \___|_.__/ \___/|___|
      |_|                                               
      
v0.1.123 - building the best open-source AI user interface.      
https://github.com/open-webui/open-webui
INFO:apps.litellm.main:start_litellm_background
INFO:apps.litellm.main:run_background_process
INFO:apps.litellm.main:Executing command: ['litellm', '--port', '14365', '--host', '127.0.0.1', '--telemetry', 'False', '--config', '/app/backend/data/litellm/config.yaml']
INFO:     Application startup complete.
INFO:     Uvicorn running on http://0.0.0.0:8080 (Press CTRL+C to quit)
INFO:apps.litellm.ma
```in:Subprocess started successfully.

**Comment by urbenlegend** [comment]
I have the same issue with the `ghcr.io/open-webui/open-webui:main` image. The models are not automatically downloaded and nothing happens when I click the audio icon underneath the chat entries. Are we supposed to download these models manually? If so, where do we get them and how do we install them? There's no documentation for this feature at all besides the environment variables.

**Comment by EgidioSuriano** [comment]
Is there documentation for being able to use Whisper in Open WebUI? Also, is it possible to upload an audio file and have the audio transcript in text directly in the prompt box?

**Comment by duskosusa** [comment]
I would like to know too. Whisper(local) appeared briefly in settings and disappeared. How do I add it back?
